src/htsjdk/java/htsjdk/tribble/index/tabix/TabixIndexMerger.java

```diff
@@ -17,58 +18,90 @@
 import java.util.List;
 
 /** Merges tabix files for parts of a file that have been concatenated. */
-public class TabixIndexMerger {
+public class TabixIndexMerger implements Closeable {
 
-  /**
-   * Merge tabix files for (headerless) file parts.
-   * @param partLengths the lengths, in bytes, of the headerless file parts
-   * @param tbiStreams streams for the tabix files to merge
-   * @param tbiOut the output stream for the resulting merged tabix file
-   */
-  public static void merge(
-      List<Long> partLengths,
-      List<SeekableStream> tbiStreams,
-      OutputStream tbiOut) throws IOException {
-    if (tbiStreams.isEmpty()) {
-      throw new IllegalArgumentException("Cannot merge zero tabix files");
-    }
-    List<TabixIndex> tbis = new ArrayList<>();
-    for (SeekableStream tbiStream : tbiStreams) {
-      tbis.add(new TabixIndex(new BlockCompressedInputStream(tbiStream)));
-    }
+  private final OutputStream out;
+  private final List<Long> partLengths;
+  private TabixFormat formatSpec;
+  private List<String> sequenceNames;
+  private List<List<BinningIndexContent>> content;
 
-    TabixFormat formatSpec = tbis.get(0).getFormatSpec();
-    List<String> sequenceNames = tbis.get(0).getSequenceNames();
-    for (TabixIndex tbi : tbis) {
-      if (!tbi.getFormatSpec().equals(formatSpec)) {
-        throw new IllegalArgumentException(
-            String.format("Cannot merge tabix files with different formats, %s and %s.", tbi.getFormatSpec(), formatSpec));
-      }
-      if (!tbi.getSequenceNames().equals(sequenceNames)) {
-        throw new IllegalArgumentException(
-            String.format("Cannot merge tabix files with different sequence names, %s and %s.", tbi.getSequenceNames(), sequenceNames));
+  public TabixIndexMerger(final OutputStream out, final long headerLength) {
+    this.out = out;
+    this.partLengths = new ArrayList<>();
+    this.partLengths.add(headerLength);
+  }
+
+  public void processIndex(TabixIndex index, long partLength) {
+    this.partLengths.add(partLength);
+    if (content == null) {
+      content = new ArrayList<>();
+      formatSpec = index.getFormatSpec();
+      sequenceNames = index.getSequenceNames();
```

Comment:
similar here, is `index.getSequenceNames()` guaranteed not to be null?  If so, the next line could NPE.  Consider making sequenceNames a final empty List.

Reply:
Done